### PR TITLE
Random Extensions and Static Random

### PR DESCRIPTION
**Pull request by amulware**
Added extension methods to Random and added static StaticRandom class with thread safe static access to pseudo random numbers.

Made a lot of changes to original comments for consistency and completeness.

### CODE REVIEW

File: src/Core/RandomExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+        /// Returns -1 or 1 randomly.
+        /// </summary>
+        /// <param name="random">The Random instance to sample with.</param>
+        public static int NextSign(this Random random)
+        {
+            return 2 * random.Next(2) - 1;
+        }
+
+        /// <summary>
+        /// Returns a random boolean value.
```
Comment (tomrijnbeek) [inline src/Core/RandomExtensions.cs]: include "with specified probability"?

File: src/Core/RandomExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        /// </summary>
+        /// <param name="random">The Random instance to sample with.</param>
+        /// <param name="probability">The probability with which to return true.
+        /// Always returns true for values greater or equal to 1, and false for values less or equal to 0.</param>
```
Comment (tomrijnbeek) [inline src/Core/RandomExtensions.cs]: Would put this line in a `<returns>` tag.

File: src/Core/StaticRandom.cs [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+        public static int Sign()
+        {
+            return random.NextSign();
+        }
+
+        /// <summary>
+        /// Returns a random boolean value.
+        /// </summary>
+        /// <param name="probability">The probability with which to return true.
+        /// Always returns true for values greater or equal to 1, and false for values less or equal to 0.</param>
```
Comment (tomrijnbeek) [inline src/Core/StaticRandom.cs]: id. from previous comments

File: src/Core/StaticRandom.cs [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+        public static bool Bool(double probability = 0.5)
+        {
+            return random.NextBool(probability);
+        }
+
+
+        /// <summary>
+        /// Returns an integer, with a given expected value. Will always return either the floor or ceil of the given value.
```
Comment (tomrijnbeek) [inline src/Core/StaticRandom.cs]: those commas o_O

**Comment by amulware** [comment]
Messed up merge... will make new pull request with proper branch name.

**Pull request closed without merging**